### PR TITLE
Add converted headers check functionality

### PR DESCRIPTION
**Pull request by gclssvglx**
### What
Add the code that calculates the list of countries - where the appropriate content headers have been converted - from this tool: https://foreign-travel-spike.herokuapp.com/.

### Why
We want to dynamically look this list up in the `check_travel_during_coronavirus` flow instead of hard-coding the list in the [countries_with_content_headers_converted](https://github.com/alphagov/smart-answers/blob/8680866ca9e181141f10a1349da332f8009be6d3/lib/smart_answer/calculators/check_travel_during_coronavirus_calculator.rb#L82) method so that it is always up-to-date.

### ⚠️ Testing locally
If you want to test the caching functionality locally in `development`, please remember to run `rails dev:cache` first. This simply creates an empty file at `tmp/caching-dev.txt`. This is a toggle switch - so create/delete. You might want to turn it back off when you're done testing as it may cause un-expected and hard-to-diagnose caching issues.

Trello: https://trello.com/c/Oq6GfWTV/665-add-converted-headers-list-functionality

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: app/services/travel_advice_header_check_retriever.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class TravelAdviceHeaderCheckRetriever
+  def self.fetch(country_slug)
+    Rails.cache.fetch(country_slug, expires_in: 4.hours) do
+      TravelAdviceHeaderChecker.new(country_slug).has_content_headers?
```
Comment (Rosa-Fox) [inline app/services/travel_advice_header_check_retriever.rb:4]: 👍

File: test/flows/check_travel_during_coronavirus_flow_test.rb [hunk 2]
```diff
@@ -6,6 +6,39 @@ class CheckTravelDuringCoronavirusFlowTest < ActiveSupport::TestCase
 
   setup do
     testing_flow CheckTravelDuringCoronavirusFlow
+
```
Comment (Rosa-Fox) [inline test/flows/check_travel_during_coronavirus_flow_test.rb:9]: wondering if this could go in a helper to make the file smaller?

**Comment by Rosa-Fox** [comment]
Hi @gclssvglx looks good! I had some questions about error handling... its quite hard to tell without encountering the errors but we could write some tests for them. The rest of the comments are small refactoring things. Let me know what you think!

**Comment by Rosa-Fox** [comment]
Also thanks for the detailed commits 🎉

File: app/models/travel_advice_header_checker.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        return Nokogiri::HTML.parse(part["body"])
+      end
+    end
+
+    nil
+  end
+
+  def has_content_header?(entry_requirements, header)
+    node = entry_requirements.at_css("[id='#{header}']")
+    node.present? && node.children.any?
```
[the remark this answers is not in the document]
Comment (gclssvglx) [reply app/models/travel_advice_header_checker.rb:47]: Well, yes - and no! 

If we have some content like this for example `<h2 id="all-travellers">All travellers</h2><p>Content</p>` what `node.children` returns is `All travellers` - so `node.children.any?` is actually testing that the header exists - which is fine for our purpose here - and that the header itself has a value - `All travellers` in this case.

If we wanted to get `Content` out of this - actually return the stuff between this header and - well, what exactly? We'd need something like this...

```ruby
node = entry_requirements.at_css("[id='#{header}']")
child_nodes(node).join("") # returns "Content"
  
def child_nodes(node)
  current_node = node.next
  content = []
  loop do
    break if current_node.nil? || ["h1", "h2"].include?(current_node.name)
    content << current_node
    current_node = current_node.next
  end
  content
end
```


This is where it all gets a tad evil!  

I'm assuming that the header node is an `H2`, so I'm looking for either:
* the next `H2` - assuming anything else in-between the current and next `H2` is 'below' the header as far as HTML is concerned, or 
* an `H1` - assuming we are going up a level, or 
* we've hit the end of the content.

I'll try and update the tests to make it more obvious what is going on.

File: app/models/travel_advice_header_checker.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    end
+
+    true
+  end
+
+private
+
+  def entry_requirements_body_content
+    response = HTTParty.get("#{base_url}/#{country_slug}", headers: headers)
+    parsed_response = JSON.parse(response.body)
```
[the remark this answers is not in the document]
Comment (gclssvglx) [reply app/models/travel_advice_header_checker.rb]: Thanks - fixed. I've added some test cases, but found some errors - particularly some of those generated by HTTParty - harder to simulate.

File: app/models/travel_advice_header_checker.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (Rosa-Fox) [reply app/models/travel_advice_header_checker.rb:74]: Thanks! Yeah just checking the header itself seems good.

File: app/models/travel_advice_header_checker.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (Rosa-Fox) [reply app/models/travel_advice_header_checker.rb]: thanks!

File: app/models/travel_advice_header_checker.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (Rosa-Fox) [reply app/models/travel_advice_header_checker.rb:74]: Thanks! Yeah just checking that the header content itself exists seems good.

**Review by Rosa-Fox — APPROVED** [review]
Looks good! Thanks for making the changes. I'll deploy in the morning.

**Pull request merged**